### DISCUSSION TITLE
关于多IDC中子集群转发的疑问

### DISCUSSION THREAD

**Discussion by fanghf0108** [bfenetworks/bfe#848] [Q&A (问题和回答)] [answered]
存在多IDC的情况下，如果IDC地理位置跨度较大，如分布在中国和海外。这种情况下跨IDC进行子集群转发的时候，也是由BFE直接转到别的IDC中的子集群吗？还是会先在BFE之间转发请求？如在IDC1有BFE1和SubCluster1、IDC2有BFE2和SubCluster2，BFE1的请求会直接打到SubCluster2吗？还是会先从BFE1转到BFE2，再由BFE2到SubCluster2？

敬待您的回答，感谢！

**Comment by mileszhang2016** [comment, marked as answer]
您这个问题应该是针对“如何实现多数据中心流量调度”这篇文章吧？
https://segmentfault.com/a/1190000040736825

如果仅在数据中心内有网络接入点，请求是由网络接入点所在数据中心内的BFE集群转发到下游的服务集群（可能位于同数据中心，也可能位于不同数据中心）。这个已经在上文中说明。在这种场景下，不会有BFE集群到BFE集群的转发。主要原因是那样做会增加资源的消耗，而且会引入额外的复杂性，而这些都不是必要的。

另外一种做法是：在数据中心之外，建立独立的“边缘节点”（Edge），在Edge内也部署BFE集群。Edge节点和数据中心有较好的网络互联条件（一般是网络专线）。用户的请求首先到达Edge的BFE集群，然后再转发给数据中心内的BFE集群。在这种情况下，会出现BFE集群的“级连”（如下图所示）。在海外没有部署服务的情况下，可以在海外部署Edge节点，再转发给数据中心。

![image](https://user-images.githubusercontent.com/16986845/135186732-b6dcf51b-b174-4877-908a-9050358bd401.png)

**Reply by fanghf0108** [reply]
了解了，非常感谢您的回答，多谢！
